_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,420 @@
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaUser
+metadata:
+  name: my-user
+  labels:
+    strimzi.io/cluster: <CLUSTER-NAME>
+spec:
+  authentication:
+    type: tls
+  authorization:
+    type: simple
+    acls:
+      - resource:
+          type: topic
+          name: my-topic-1
+          patternType: literal
+        operation: Write
+        host: "*"
+      - resource:
+          type: topic
+          name: my-topic-1
+          patternType: literal
+        operation: Read
+        host: "*"
+      - resource:
+          type: topic
+          name: my-topic-1
+          patternType: literal
+        operation: Describe
+        host: "*"   
+      - resource:
+           type: topic
+           name: my-topic-1
+           patternType: literal
+        operation: DescribeConfigs
+        host: "*"
+      - resource:
+           type: topic
+           name: my-topic-1
+           patternType: literal
+        operation: AlterConfigs
+        host: "*"        
+      - resource:
+           type: cluster
+           name: <CLUSTER-NAME>
+           patternType: literal
+        operation: Alter
+        host: "*"
+      - resource:
+           type: cluster
+           name: <CLUSTER-NAME>
+           patternType: literal
+        operation: AlterConfigs
+        host: "*"        
+  # ...
+```
+
+### Creating a proposal `reassignment.json` file
+
+When you have a Kafka cluster running with brokers and topics, you are ready to create the proposal JSON file.
+The fist step is to extract the CA certificates and keys for the Kafka cluster and the user in order to authenticate using TLS client authentication.
+
+We can easily extract the cluster CA certificate and password from the `<CLUSTER-NAME>-cluster-ca-cert` Secret (where `<CLUSTER-NAME>` denotes the name of the cluster) using the following commands:
+
+```sh
+kubectl get secret CLUSTER-NAME-cluster-ca-cert -o jsonpath='{.data.ca\.p12}' | base64 -d > ca.p12
+```
+
+```sh
+kubectl get secret CLUSTER-NAME-cluster-ca-cert -o jsonpath='{.data.ca\.password}' | base64 -d > ca.password
+```
+
+Let us create a separate interactive pod.
+This interactive pod is used to run all the reassignment commands. 
+One question might bug you. What is the need of a separate pod? Can't we just use one of the broker pods?
+In answer to this question, running commands from within a broker is not good practice as it's insecure
+It will start another JVM inside the container designated for the broker and can cause disruption, cause the container to run out of memory and so on.
+So it is always better to avoid running the command from a broker pod.
+
+So now it's time to get our interactive pod up and running. You can use the following command:
+
+```sh
+kubectl run --restart=Never --image=quay.io/strimzi/kafka:0.27.0-kafka-3.0.0 <interactive_pod_name> -- /bin/sh -c "sleep 3600"
```

Comment:
You now use everywhere `<...>` for the values which should be replaced. But the content is still different:
* Uppercase versus lowercase
* `-` versus `_`